kotlinpoet/src/commonMain/kotlin/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -390,12 +390,16 @@ public class TypeSpec private constructor(
         parameter.kdoc != propertyKdoc
     }
     return with(kdoc.ensureEndsWithNewLine().toBuilder()) {
-      if (kdoc.isNotEmpty()) add("\n")
-      if (primaryConstructor.kdoc.isNotEmpty()) {
-        add("@constructor %L", primaryConstructor.kdoc.ensureEndsWithNewLine())
-      }
-      parametersWithKdoc.forEach { parameter ->
-        add("@param %L %L", parameter.name, parameter.kdoc.ensureEndsWithNewLine())
+      if (primaryConstructor.kdoc.isNotEmpty() || parametersWithKdoc.isNotEmpty()) {
+        if (isNotEmpty()) {
+          add("\n")
+        }
+        if (primaryConstructor.kdoc.isNotEmpty()) {
+          add("@constructor %L", primaryConstructor.kdoc.ensureEndsWithNewLine())
+        }
+        parametersWithKdoc.forEach { parameter ->
+          add("@param %L %L", parameter.name, parameter.kdoc.ensureEndsWithNewLine())
+        }
       }
       build()
     }
```

Comment:
I wanted to avoid a thousand `if`s so did this, see if you like it better (sadly can't add as a suggestion):

```kotlin
val constructorKdoc = buildCodeBlock {
  if (primaryConstructor.kdoc.isNotEmpty()) {
    add("@constructor %L", primaryConstructor.kdoc.ensureEndsWithNewLine())
  }
  parametersWithKdoc.forEach { parameter ->
    add("@param %L %L", parameter.name, parameter.kdoc.ensureEndsWithNewLine())
  }
}
return listOf(kdoc.ensureEndsWithNewLine(), constructorKdoc)
  .filter(CodeBlock::isNotEmpty)
  .joinToCode(separator = "\n")
```

Reply:
Ship it

Reply:
Ended up pulling even more into this. Because the code above also had conditionals and functional list operations. Combined everything as best I could with minimal conditional logic.